src/pages/admin/HeatMapPage.tsx

```diff
@@ -0,0 +1,303 @@
+import React, { useMemo, useState, useEffect } from 'react';
+import {
+  Box,
+  Tooltip,
+  Grid,
+  GridItem,
+  VStack,
+  Text,
+  Spinner,
+  Center,
+  HStack,
+  useBreakpointValue,
+  Input,
+  Button,
+} from '@chakra-ui/react';
+import { useHeatmapColor } from '../../hooks/useHeatmapColor';
+import { TEN_MINUTES, DAY, formatDate } from '../../localization';
+import { getSignupTimeline } from '../../services/interview';
+import { RawSignup } from '../../types';
+import { useNavigate } from 'react-router-dom';
+import { ArrowBackIcon } from '@chakra-ui/icons';
+import { AxiosError } from 'axios';
+
+const INCREMENTS_PER_DAY = 144;
+
+const tsForBucket = (bucket: number) => bucket * TEN_MINUTES;
+
+interface DayTimelineProps {
+  signups: RawSignup[];
+  selectedDay: number;
+  height: number;
+}
+
+const SignupsDetails: React.FC<{
+  signups: RawSignup[];
+  bucketIdx: number;
+  startTime: number;
+}> = ({ signups, bucketIdx, startTime }) => {
+  const bucketStart = tsForBucket(bucketIdx) + startTime;
+  const bucketEnd = tsForBucket(bucketIdx + 1) + startTime;
+
+  const signupsInBucket = signups.filter(
+    ({ timestamp }) => timestamp >= bucketStart && timestamp < bucketEnd
+  );
+  return (
+    <Box>
+      {signupsInBucket.map(({ timestamp, username, user_id }) => (
+        <Text key={`${timestamp}-${user_id}`} fontSize="sm">
+          {formatDate(timestamp, true, true)} - {username} ({user_id})
+        </Text>
+      ))}
+    </Box>
+  );
+};
+
+const DayTimeline: React.FC<DayTimelineProps> = ({
```

Comment:
Maybe separate out each component into its own file in the components directory? You can put it in a directory called `heatmap` if you want to organize by related components